### PR TITLE
Merge API_migration and v2-migration branches

### PR DESCRIPTION
**Pull request by rjouhann**
**⚠️ THIS PR WILL BE MERGE ALONG WITH THE MARCH SELF-HOSTED RELEASE ⚠️** 

This MR merges the `API_migration` branch and the `v2-migration` branch into `main`.

I am also updating the readme for the root directory and the v2-migration.

### CODE REVIEW

File: new-arch-migration/EMBEDDED.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+GitGuardian provides a set of scripts that require specific tools to be installed on your host to facilitate application migration:
+
+- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) (version ≥ 1.27.0)
+- [kubectl kots plugin](https://docs.replicated.com/reference/kots-cli-getting-started#install) (version ≥ 1.107.7)
+- [yq](https://mikefarah.gitbook.io/yq/) (Only for Blue/Green Migration)
+
+You need to be an administrator of the GitGuardian namespace where the application is deployed.
+
+The new version must use the same GitGuardian version as the legacy version. Please ensure you have the latest legacy version installed before upgrading to the new version.
```
Comment (elacaille18) [inline new-arch-migration/EMBEDDED.md:13]: This sentence looks a bit confusing: **version** refers to both
- the architecture 
- the self-hosted release.

Maybe use the **architecture** word?

File: new-arch-migration/README.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
+# Migrating from the legacy to the new Architecture
+
+To initiate the migration process or if you have any questions regarding the new architecture, please reach out to our [support team](mailto:support@gitguardian.com?subject=Migration+New+Architecture). We aim for a smooth migration process to enhance your GitGuardian setup with a future-ready, secure, and scalable architecture that aligns with the demands of modern cloud environments.
+
+Explore the [New Architecture documentation](https://docs.gitguardian.com/self-hosting/new-architecture) for a deep dive into its advantages, including enhanced performance, security, and scalability features. Our guide provides a thorough understanding of the architectural upgrades and the additional benefits of switching to the new version of GitGuardian.
```
Comment (elacaille18) [inline new-arch-migration/README.md:5]:
```suggestion
Explore the [New Architecture documentation](https://docs.gitguardian.com/self-hosting/new-architecture) for a deep dive into its advantages, including enhanced performance, security, and scalability features. Our guide provides a thorough understanding of the architectural upgrades and the additional benefits of switching to the new architecture of GitGuardian.
```

🤔

File: new-arch-migration/README.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+GitGuardian provides a set of scripts that require specific tools to be installed on your host to facilitate application migration:
+
+- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) (version ≥ 1.27.0)
+- [kubectl kots plugin](https://docs.replicated.com/reference/kots-cli-getting-started#install) (version ≥ 1.107.7)
+- [yq](https://mikefarah.gitbook.io/yq/) (Only for Blue/Green Migration)
+
+You need to be an administrator of the GitGuardian namespace where the application is deployed.
+
+The new version must use the same GitGuardian version as the legacy version. Please ensure you have the latest legacy version installed before upgrading to the new version.
```
Comment (elacaille18) [inline new-arch-migration/README.md:32]: same comment as above (for the same sentence) 😉

File: new-arch-migration/README.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+
+⚠️ The GitGuardian team needs to update your license information (Channel switching from `prod` to `stable`) to provide you with the new version of the application, so you need to [sync with them](?subject=Migration+New+Architecture+in+place+migration+external) before upgrading.
```
Comment (elacaille18) [inline new-arch-migration/README.md:34]:
```suggestion
⚠️ The GitGuardian team needs to update your license information (Channel switching from `prod` to `stable`) to provide you with the new architecture of the application, so you need to [sync with them](?subject=Migration+New+Architecture+in+place+migration+external) before upgrading.
```
🤔

File: new-arch-migration/README.md [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+## In-place migration with external databases
+
+⚠️ Please note, this migration guide is specifically designed for customers who have installed GitGuardian on an [existing Kubernetes cluster with an external database](https://docs.gitguardian.com/self-hosting/installation/installation-existing-cluster-legacy). If your GitGuardian instance is running on an embedded cluster, visit this [page](./EMBEDDED.md).
+
+⚠️ This migration will require some downtime, which may take up to one hour.
+
+ℹ️ For airgap installation, first, download the airgap bundle file from your download portal.
+
+1. To begin with, please create a backup of your GitGuardian's external PostgreSQL database.
+2. You can now migration GitGuardian to the new architecture using the following command line:
```
Comment (elacaille18) [inline new-arch-migration/README.md:45]:
```suggestion
2. You can now migrate GitGuardian to the new architecture using the following command line:
```
💅

File: new-arch-migration/TOPOLOGY.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
+## Application Topology
+
+Here's a side-by-side table comparing the new GitGuardian application container architecture with the legacy version. This table highlights the changes in deployment names and their usage between the two versions. You can also find architecture diagrams on [this page](https://gg-code.pages.gitguardian.ovh/public-docs/jouhannet-release-2024-3-0/self-hosting/new-architecture#a-more-scalable-architecture).
```
Comment (elacaille18) [inline new-arch-migration/TOPOLOGY.md:3]:
```suggestion
Here's a side-by-side table comparing the new GitGuardian application container architecture with the legacy architecture. This table highlights the changes in deployment names and their usage between the two architecture. You can also find architecture diagrams on [this page](https://gg-code.pages.gitguardian.ovh/public-docs/jouhannet-release-2024-3-0/self-hosting/new-architecture#a-more-scalable-architecture).
```
🤔

**Review by elacaille18 — APPROVED** [review]
Small 💅  comments in order to avoid confusion between **version** and **architecture**

**Review by wasshana-gg — COMMENTED** [review]
I don't have anything to add, it's genuinely very easy to follow as a procedure. I recommended an additional step to the procedure just to be on the safe side.

**Pull request merged**